src/Categories/Bicategory/Construction/Spans.agda

```diff
@@ -0,0 +1,406 @@
+{-# OPTIONS --without-K --safe #-}
+
+open import Categories.Category
+open import Categories.Diagram.Pullback
+
+module Categories.Bicategory.Construction.Spans {o ℓ e} (𝒞 : Category o ℓ e) (has-pullbacks : ∀ {X Y Z} (f : 𝒞 [ X , Z ]) (g : 𝒞 [ Y , Z ]) → Pullback 𝒞 f g) where
+
+open import Level
+open import Function using (_$_)
+
+open import Data.Product using (Σ; _×_; _,_; proj₁; proj₂; uncurry)
+
+open import Categories.Bicategory
+open import Categories.NaturalTransformation.NaturalIsomorphism hiding (refl; sym)
+
+open import Categories.Category.Construction.Spans 𝒞 renaming (Spans to Spans₁)
+
+open import Categories.Morphism
+open import Categories.Morphism.Reasoning 𝒞
+
+private
+  module Spans₁ X Y = Category (Spans₁ X Y)
+
+open Category 𝒞
+open HomReasoning
+open Equiv
+
+open Span
+open Span⇒
+open Pullback
+
+private
+  variable
+    A B C D E : Obj
+
+  --------------------------------------------------------------------------------
+  -- Various Spans, Span Maps, and their compositions
+  
+  id-span : Span A A
+  id-span {A} = record
+    { M = A
+    ; dom = id
+    ; cod = id
+    }
+  
+  _⊚₀_ : Span B C → Span A B → Span A C
+  f ⊚₀ g =
+    let pullback = has-pullbacks (cod g) (dom f)
+    in record
+        { M = P pullback
+        ; dom = dom g ∘ p₁ pullback
+        ; cod = cod f ∘ p₂ pullback
+        }
+  
+  _⊚₁_ : {f f′ : Span B C} {g g′ : Span A B} → Span⇒ f f′ → Span⇒ g g′ → Span⇒ (f ⊚₀ g) (f′ ⊚₀ g′)
+  _⊚₁_  {_} {_} {_} {f} {f′} {g} {g′} α β =
+    let pullback = has-pullbacks (cod g) (dom f)
+        pullback′ = has-pullbacks (cod g′) (dom f′)
+    in record
+      { arr = universal pullback′ {h₁ = arr β ∘ p₁ pullback} {h₂ = arr α ∘ p₂ pullback} $ begin
+        cod g′ ∘ arr β ∘ p₁ pullback ≈⟨ pullˡ (commute-cod β) ⟩
+        cod g ∘ p₁ pullback          ≈⟨ commute pullback ⟩
+        dom f ∘ p₂ pullback          ≈⟨ pushˡ (⟺ (commute-dom α)) ⟩
+        dom f′ ∘ arr α ∘ p₂ pullback ∎
+      ; commute-dom = begin
+        (dom g′ ∘ p₁ pullback′) ∘ universal pullback′ _ ≈⟨ pullʳ (p₁∘universal≈h₁ pullback′) ⟩
+        dom g′ ∘ arr β ∘ p₁ pullback                    ≈⟨ pullˡ (commute-dom β) ⟩
+        dom g ∘ p₁ pullback                             ∎
+      ; commute-cod = begin
+        (cod f′ ∘ p₂ pullback′) ∘ universal pullback′ _ ≈⟨ pullʳ (p₂∘universal≈h₂ pullback′) ⟩
+        cod f′ ∘ arr α ∘ p₂ pullback                    ≈⟨ pullˡ (commute-cod α) ⟩
+        cod f ∘ p₂ pullback                             ∎
+      }
+  
+  --------------------------------------------------------------------------------
+  -- Proofs about span compositions
```

Comment:
There proofs should probably be defined as properties of `Span` rather than 'buried' here.

Reply:
Similar to the two composition primitives, all of these properties rely on the existence of pullbacks, which makes them a bit awkward to use separate from the bicategory instance